### PR TITLE
gcc version check for pragma diagnostic

### PR DESCRIPTION
**Pull request by LEvertz**
the diagnostic push / pop pragma is new since gcc version 4.6
this pull adds a test for the gcc version to avoid errors in older versions

### CODE REVIEW

File: examples/logger_stdout.c [hunk 1]
```diff
@@ -18,7 +18,7 @@ static void print_time(void) {
 const char *LogLevelNames[6] = {"trace", "debug", "info", "warning", "error", "fatal"};
 const char *LogCategoryNames[4] = {"communication", "server", "client", "userland"};
 
-#if defined(__GNUC__) || defined(__clang__)
+#if ((__GNUC__ >= 4 && __GNUC_MINOR__ >= 6) || defined(__clang__))
```
Comment (HolgerJeromin) [inline examples/logger_stdout.c]: This would not match for example gcc 5.3

Comment (LEvertz) [inline examples/logger_stdout.c]: Right, my bad.

I’ll fix it and recommit

**Comment by Stasik0** [comment]
Is it tested for an older gcc? I suspect that we need to turn "treat warnings as errors" off if we cannot use pragma statements.

**Comment by Stasik0** [comment]
I meant the -Werror flag

**Comment by LEvertz** [comment]
It seemed to work anyway on my gcc (4.5.3)

(at first with out-commented pragmas)

**Comment by Stasik0** [comment]
@LEvertz i have pulled and squashed your commits, please synch your branch

**Pull request closed without merging**